### PR TITLE
cohttp-eio: migrate eio tests to mdx based tests

### PR DESCRIPTION
**Pull request by bikallem**
This PR updates cohttp-eio tests to use mdx based tests primarily using `eio.mock` package. This allows tests to be repeatable especially with regards to networks and system time facilities.

### CODE REVIEW

**Comment by bikallem** [comment]
@mseri The PR is ready for review now. It is mostly tests changes so please advise if the PR needs a Changelog.

File: cohttp-eio/src/client.ml [hunk 1]
```diff
@@ -44,7 +44,8 @@ let write_request request writer body =
   Buf_write.string writer "\r\n";
   Rwer.write_headers writer headers;
   Buf_write.string writer "\r\n";
-  Body.write_body ~write_chunked_trailers:true writer body
+  Body.write_body ~write_chunked_trailers:true writer body;
+  Buf_write.flush writer
```
Comment (mseri) [inline cohttp-eio/src/client.ml:48]: Can you add this in a separate commit, and mention on the the fix in the changelog?

Comment (bikallem) [reply cohttp-eio/src/client.ml:48]: Done. Moved this bit to a separate PR (https://github.com/mirage/ocaml-cohttp/pull/949) - which needs to be merged before this one.

**Comment by bikallem** [comment]
@talex5 @mseri if there isn't anything outstanding or blocking issues could we merge this PR. I have an upcoming PR (date header related) that uses the mdx/mock based test here.

**Comment by mseri** [comment]
It is still not merged because I need a bit of time to actually review it, even though it is just moving tests around

File: cohttp-eio/tests/client.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+     `Raise End_of_file
+    ]
+    ~test:(fun () ->
+      (Client.get
+        ~headers:(Http.Header.of_list [ ("Accept", "application/json") ])
+        ~conn
+        host
+        "/")
+      |> Client.read_fixed
+      |> ignore);;
```
Comment (talex5) [inline cohttp-eio/tests/client.md:39]: Why are you ignoring the result? This isn't testing anything except that the client read from the socket!

(this comment applies to the other tests in this file too)

Comment (bikallem) [reply cohttp-eio/tests/client.md]: > Why are you ignoring the result? This isn't testing anything except that the client read from the socket!

The tests in client.md mostly tests the `Cohttp_eio.Client` response headers and that we can successfully receive response without any exceptions/errors. However, yes it does seem wasteful to ignore the `read_fixed` results. I have modified the tests to check for valid response body where possible.

**Comment by mseri** [comment]
Can you rebase on master and for e push? If tests are green I will merge

**Comment by bikallem** [comment]
Thanks.

**Pull request merged**